fea-rs/src/compile/output.rs

```diff
@@ -55,18 +52,25 @@ impl Compilation {
     /// and manipulate the raw tables directly.
     ///
     /// [`to_binary`]: Compilation::to_binary
-    pub fn to_font_builder(&self) -> Result<FontBuilder, BinaryCompilationError> {
+    pub fn to_font_builder(&self) -> Result<FontBuilder, BuilderError> {
         let mut builder = FontBuilder::default();
-        builder.add(self.head.as_ref())?;
-        builder.add(self.hhea.as_ref())?;
-        builder.add(self.vhea.as_ref())?;
-        builder.add(self.os2.as_ref())?;
-        builder.add(self.gdef.as_ref())?;
-        builder.add(self.base.as_ref())?;
-        builder.add(self.name.as_ref())?;
-        builder.add(self.stat.as_ref())?;
-        builder.add(self.gsub.as_ref())?;
-        builder.add(self.gpos.as_ref())?;
+        macro_rules! add_if_some {
```

Comment:
I believe we’ve established a general preference for binding a closure to a local rather than using a macro in cases like this. I think the code should be roughly equivalent?

Reply:
I don't think it works, because each of these tables is a different concrete type, and i don't think I can have generics in a closure?

Reply:
going to merge as-is, if it turns out there's a better approach we can update as needed :)